scipy/spatial/procrustes.py

```diff
@@ -0,0 +1,176 @@
+        matrix, n rows represent points in k (columns) space data1 is the
+        reference data, after it is standardised, the data from data2 will
+        be transformed to fit the pattern in data1 (must have >1 unique
+        points).
+
+    data2 : array_like
+        n rows of data in k space to be fit to data1.  Must be the  same
+        shape (numrows, numcols) as data1 (must have >1 unique points).
+
+    Returns
+    -------
+    mtx1 : array_like
+        a standardized version of data1
+    mtx2 : array_like
+        the orientation of data2 that best fits data1. Centered, but not
+        necessarily :math:`tr(AA^{T}) = 1`.
+    disparity : float
+        :math:`M^{2}` as defined above.
+
+    See Also
+    --------
+    orthogonal_procrustes
+
+    Notes
+    -----
+    - The disparity should not depend on the order of the input matrices, but
+      the output matrices will, as only the first output matrix is guaranteed
+      to be scaled such that :math:`tr(AA^{T}) = 1`.
+
+    - Duplicate data points are generally ok, duplicating a data point will
+      increase its effect on the procrustes fit.
+
+    - The disparity scales as the number of points per input matrix.
+
+    References
+    ----------
+    .. [1] Krzanowski, W. J. (2000). "Principles of Multivariate analysis".
+    .. [2] Gower, J. C. (1975). "Generalized procrustes analysis".
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from scipy.spatial import procrustes
+    # the b matrix is a rotated, shifted, scaled and mirrored version of a
+    >>> a = np.array([[1, 3], [1, 2], [1, 1], [2, 1]], 'd')
+    >>> b = np.array([[4, -2], [4, -4], [4, -6], [2, -6]], 'd')
+    >>> mtx1, mtx2, disparity = procrustes(a, b)
+    >>> round(disparity)
+    0.0
+
+    """
+    num_rows, num_cols = np.shape(data1)
+    if (num_rows, num_cols) != np.shape(data2):
+        raise ValueError("Input matrices must be of same shape")
+    if num_rows == 0 or num_cols == 0:
+        raise ValueError("Input matrices must be >0 rows and >0 cols")
+
+    # standardize each matrix
+    mtx1 = _center(data1)
+    mtx2 = _center(data2)
+
+    if (not np.any(mtx1)) or (not np.any(mtx2)):
+        raise ValueError("input matrices must contain >1 unique points")
+
+    mtx1 = _normalize(mtx1)
+    mtx2 = _normalize(mtx2)
+
+    # transform mtx2 to minimize disparity (sum( (mtx1[i,j] - mtx2[i,j])^2) )
+    mtx2 = _match_points(mtx1, mtx2)
+
+    disparity = _get_disparity(mtx1, mtx2)
+
+    return mtx1, mtx2, disparity
+
+
+def _center(mtx):
+    """Translate all data (rows of the matrix) to center on the origin
+
+    Parameters
+    ----------
+    mtx : array_like
+        Matrix to translate the data for.
+
+    Returns
+    -------
+    result : array_like ('d') array
+        Shifted version of the input data.  The new matrix is such that the
+        center of mass of the row vectors is centered at the origin.
+
+    """
+    result = np.array(mtx, 'd')
```

Comment:
Is it necessary for this to only support float64? I suppose float32 is the only other viable choice given the LAPACK calls, but why not support it if possible?